fc/code_generation.py

```diff
@@ -198,7 +198,7 @@ def derivative_name(deriv):
     for pvar in protocol_variables:
         if pvar.is_input and pvar.model_variable is not None:
             eq = model.get_definition(pvar.model_variable)
-            if eq is not None and not eq.lhs.is_Derivative and len(eq.rhs.atoms(VariableDummy)) == 0:
+            if eq is not None and not eq.lhs.is_Derivative and model.is_constant(pvar.model_variable):
```

Comment:
I meant to replace these lines with
```python
if pvar.is_input and pvar.model_variable is not None and model.is_constant(pvar.model_variable):
```
We don't need `eq` at all except for the constant check.